DataFormats/ParticleFlowCandidate/interface/PFCandidate.h

```diff
@@ -423,6 +424,12 @@ namespace reco {
     /// \set the timing information
     void setTime(float time, float timeError = 0.f) { time_ = time; timeError_ = timeError; }
 
+    /// fraction of hcal energy at a given depth  (depth = 1 .. 7)
+    float hcalDepthEnergyFraction(unsigned int depth) const { return hcalDepthEnergyFractions_[depth-1]; }
+    /// fraction of hcal energy at a given depth (index 0..6 for depth 1..7)
+    const std::array<float,7> & hcalDepthEnergyFractions() const { return hcalDepthEnergyFractions_; }
```

Comment:
how can one distinguish a zero in e.g. depth 4 for cases with having no depth 4 at all and having 0 energy contribution there while some is still possible?

Also, what would be the meaning of this variable, if any for HGCAL?

I'm pleasantly surprised that root can now persist std::array .
@Dr15Jones do you know starting from which version do we have this feature?

Reply:
I think that a vector here may be more appropriate,
even though it's less optimal

Reply:
I do not know which ROOT version first gained the ability. I'd guess it was 6.10.